pkg/server/status.go

```diff
@@ -2218,18 +2173,40 @@ func (s *statusServer) HotRangesV2(
 						log.Warningf(ctx, "cannot decode tableID for range descriptor: %s. %s", r.Desc.String(), err.Error())
 						continue
 					}
-					parent := rangeReportMetas[tableID].parentID
-					if parent != 0 {
-						tableName = rangeReportMetas[tableID].tableName
-						dbName = rangeReportMetas[parent].dbName
-					} else {
-						dbName = rangeReportMetas[tableID].dbName
-					}
-					schemaParent := rangeReportMetas[tableID].schemaParentID
-					schemaName = rangeReportMetas[schemaParent].schemaName
-					_, _, idxID, err := s.sqlServer.execCfg.Codec.DecodeIndexPrefix(r.Desc.StartKey.AsRawKey())
-					if err == nil {
-						indexName = rangeReportMetas[tableID].indexNames[idxID]
+					if err := s.sqlServer.distSQLServer.CollectionFactory.Txn(
+						ctx, s.sqlServer.internalExecutor, s.db,
+						func(ctx context.Context, txn *kv.Txn, col *descs.Collection) error {
+							commonLookupFlags := tree.CommonLookupFlags{
```

Comment:
1) we don't have such a function sadly. There's some different specializations around which are similar, but not quite the same. We've got ideas on how we'd like to couple the table to its parents moving forward, but I'm not ready for that generalization here. 

May I request that you 
1) Make a helper
2) Just return the error from the helper and filter it the call to the helper
3) Consider building yourself a cache of the ID to name mappings with a map?
  * You can define a struct just in the scope of the function to store the fields of the name.
     Alternatively, you can make yourself an nstree.MutableCatalog and just store the cached descriptors you fetch in there and pass it down to your helper? 
     
 I think you should also store the fact that you fail for an ID. It'd be a bummer if there were a bunch of data corresponding to some ID that no longer exists for whatever reason. It'd be a bummer to go to the key-value store to resolve it for each range.